File: p2.go
package main

import "fmt"

func main() {
	var fib [50]int
	var sum int
	fib[1]= 1
	fib[2]= 2
	for i:=3; i < 50; i++ {
		fib[i] = fib[i-2] + fib[i-1]
	}
	for i:=0; i <50; i++ {
		if (fib[i] % 2 == 0) && (fib[i] < 4000000){
			sum+=fib[i]
		}
	}
	fmt.Println(sum)
}

File: p02/p2.rs
fn main() {

    let mut a = 1;
    let mut b = 2;
    let mut c = 0;
    let mut sum = 2;

    while true {
        c = a + b;
        if c >= 4_000_000 {
            break;
        }
        if c % 2 == 0 {
            sum += c;
        }
        a = b;
        b = c;
    }
        
    println!("{}", sum);
}
